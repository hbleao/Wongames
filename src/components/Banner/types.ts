export type ImageProps = {
  src: string;
};

export type BannerProps = {
  image: string;
  title: string;
  subtitle: string;
  buttonLabel: string;
  buttonLink: string;
};
