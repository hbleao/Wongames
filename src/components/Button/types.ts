import React, {
  AnchorHTMLAttributes,
  ButtonHTMLAttributes,
  ReactNode
} from 'react';

type Size = 'small' | 'medium' | 'large';
type ButtonTypes =
  | AnchorHTMLAttributes<HTMLAnchorElement>
  | ButtonHTMLAttributes<HTMLButtonElement>;

export type ButtonProps = {
  size?: Size;
  fullWidth?: boolean;
  icon?: ReactNode;
  as?: React.ElementType;
} & ButtonTypes;

export type WrapperProps = {
  fullWidth?: boolean;
  size?: Size;
  hasIcon: boolean;
};
