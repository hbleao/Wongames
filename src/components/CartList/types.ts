import { GameItemProps } from 'components/GameItem/types';

export type CartListProps = {
  items: GameItemProps[];
  total: string;
};
