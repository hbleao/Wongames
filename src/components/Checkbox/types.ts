import { InputHTMLAttributes } from 'react';

export type CheckboxProps = {
  labelText?: string;
  labelFor?: string;
  isChecked?: boolean;
  labelColor?: 'black' | 'white';
  value?: string | ReadonlyArray<string> | number;
  onCheck?: (status: boolean) => void;
} & InputHTMLAttributes<HTMLInputElement>;

export type LabelProps = {
  labelColor?: 'black' | 'white';
};
