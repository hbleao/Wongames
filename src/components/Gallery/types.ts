export type GalleryImageProps = {
  src: string;
  alt: string;
  label: string;
};

export type GalleryProps = {
  items: GalleryImageProps[];
};

export type ModalProps = {
  isOpen: boolean;
};
