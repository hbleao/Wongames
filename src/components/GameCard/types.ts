import { RibbonColors, RibbonSizes } from 'components/Ribbon/types';

export type GameCardProps = {
  title: string;
  developer: string;
  image: string;
  price: string;
  promotionalPrice?: string;
  favorite?: boolean;
  ribbon?: string;
  ribbonSize?: RibbonSizes;
  ribbonColor?: RibbonColors;
  onFav: () => void;
};

export type PriceProps = {
  isPromotional?: boolean;
};
