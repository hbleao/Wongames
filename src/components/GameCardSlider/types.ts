import { GameCardProps } from 'components/GameCard/types';

export type GameCardSliderProps = {
  items: GameCardProps[];
  color?: 'white' | 'black';
};

export type WrapperProps = {
  color?: 'white' | 'black';
};
