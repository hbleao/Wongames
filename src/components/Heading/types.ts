import { ReactNode } from 'react';

export type LineColors = 'primary' | 'secondary';

export type HeadingProps = {
  children: ReactNode;
  color?: 'white' | 'black';
  lineLeft?: boolean;
  lineBottom?: boolean;
  size?: 'small' | 'medium';
  lineColor?: LineColors;
};
