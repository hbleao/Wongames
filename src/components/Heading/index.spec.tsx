import React from 'react';
import { screen } from '@testing-library/react';

import { Heading } from '.';
import { HeadingProps } from './types';
import { renderWithTheme } from 'utils/tests/renderWithTheme';

type MakeSutProps = HeadingProps;

const makeSut = ({
  children,
  color,
  lineBottom,
  lineLeft,
  size,
  lineColor
}: MakeSutProps) => {
  const sut = renderWithTheme(
    <Heading
      color={color}
      lineBottom={lineBottom}
      lineLeft={lineLeft}
      size={size}
      lineColor={lineColor}
    >
      {children}
    </Heading>
  );

  return {
    sut
  };
};

describe('Components/Heading', () => {
  it('should be render a white label', () => {
    const text = 'Most populars';
    const { sut } = makeSut({ children: text });

    const heading = screen.getByRole('heading', { name: text });

    expect(heading).toBeInTheDocument();
    expect(heading).toHaveStyle({
      color: '#FAFAFA'
    });
    expect(sut.container).toMatchSnapshot();
  });

  it('should render a black heading when color is passed', () => {
    const text = 'Most populars';
    makeSut({ children: text, color: 'black' });

    const heading = screen.getByRole('heading', { name: text });

    expect(heading).toHaveStyle({
      color: '#030517'
    });
  });

  it('should render a heading with a line to the left side', () => {
    const text = 'Most populars';
    makeSut({ children: text, lineLeft: true, lineColor: 'secondary' });

    const heading = screen.getByRole('heading', { name: text });

    expect(heading).toHaveStyle({
      'border-left': '0.7rem solid #3CD3C1'
    });
  });

  it('should render a heading with a line at the bottom', () => {
    const text = 'Most populars';
    makeSut({ children: text, lineBottom: true });

    const heading = screen.getByRole('heading', { name: text });

    expect(heading).toHaveStyleRule('border-bottom', '0.5rem solid #F231A5', {
      modifier: '::after'
    });
  });

  it('should render a heading with small size', () => {
    const text = 'Most populars';
    makeSut({ children: text, lineBottom: true, size: 'small' });

    const heading = screen.getByRole('heading', { name: text });

    expect(heading).toHaveStyle({
      'font-size': '1.6rem'
    });
    expect(heading).toHaveStyleRule('width', '3rem', {
      modifier: '::after'
    });
  });

  it('should render a heading with huge size', () => {
    const text = 'Most populars';
    makeSut({ children: text, lineBottom: true, size: 'huge' });

    const heading = screen.getByRole('heading', { name: text });

    expect(heading).toHaveStyle({
      'font-size': '5.2rem'
    });
  });

  it('should render a Heading with a secondary line color', () => {
    const text = 'Most populars';
    makeSut({
      children: text,
      lineBottom: true,
      lineLeft: true,
      lineColor: 'secondary'
    });

    const heading = screen.getByRole('heading', { name: text });
    expect(heading).toHaveStyle({ 'border-left': '0.7rem solid #3CD3C1' });
    expect(heading).toHaveStyleRule('border-bottom', '0.5rem solid #3CD3C1', {
      modifier: '::after'
    });
  });
});
