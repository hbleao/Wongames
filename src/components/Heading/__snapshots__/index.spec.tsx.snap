// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Heading should be render a white label 1`] = `
.c0 {
  color: #FAFAFA;
  font-size: 2.0rem;
}

@media (min-width:768px) {
  .c0 {
    font-size: 2.8rem;
  }
}

<div>
  <h2
    class="c0"
    color="white"
  >
    Most populars
  </h2>
</div>
`;
